spyder/widgets/findreplace.py

```diff
@@ -527,23 +528,34 @@ def replace_find(self, focus_replace_text=False, replace_all=False):
                     # Identify wrapping even when the replace string
                     # includes part of the search string
                     wrapped = True
+
                 if wrapped:
-                    if position1 == position or \
-                       is_position_sup(position1, position):
+                    if (position1 == position
+                            or is_position_sup(position1, position)):
                         # Avoid infinite loop: replace string includes
                         # part of the search string
                         break
+
                 if position1 == position0:
                     # Avoid infinite loop: single found occurrence
                     break
                 position0 = position1
+
             if re_pattern is None:
                 cursor.removeSelectedText()
                 cursor.insertText(replace_text)
             else:
                 seltxt = to_text_string(cursor.selectedText())
+
+                # Note: If the selection obtained from an editor spans a line
+                # break, the text will contain a Unicode U+2029 paragraph
+                # separator character instead of a newline \n character.
+                # See: spyder-ide/spyder#2675
+                seltxt = seltxt.replace(u'\u2029', u'\n')

```

Comment:
Should we use here the EOL characters used in the file, instead of `\n`?

Reply:
Hmmmm good point. Are the EOF chars handled in real time or  at the moment of saving the file?

Reply:
I think you could use the function `get_eol_chars` (present on `utils/sourcecode.py`) on `seltxt` to get them.